cogs/add_users_to_threads_and_channels.py

```diff
@@ -0,0 +1,310 @@
+            for user_or_role in users_or_roles:
+                await self.add_users_or_roles_silently(
+                    users_or_roles=user_or_role, thread=thread
+                )
+            return
+
+        message: discord.Message = await thread.send(
+            content=f"Adding {users_or_roles!r} to thread...",
+            silent=True,
+        )
+        await message.edit(content=f"{users_or_roles.mention}")
+        await message.delete(delay=1)
+
+    async def add_users_or_roles_with_ping(
+        self,
+        users_or_roles: discord.Member
+        | discord.Role
+        | list[discord.Member]
+        | list[discord.Role],
+        thread: discord.Thread,
+    ) -> None:
+        """Add a user or role to a thread and ping them."""
+        if isinstance(users_or_roles, Iterable):
+            user_or_role: discord.Role | discord.Member
+            for user_or_role in users_or_roles:
+                await self.add_users_or_roles_with_ping(
+                    users_or_roles=user_or_role, thread=thread
+                )
+            return
+
+        if isinstance(users_or_roles, discord.Member):
+            try:
+                await thread.add_user(user=users_or_roles)
+            except discord.NotFound:
+                logger.debug(
+                    "User: %s has blocked the bot and "
+                    "therefore could not be added to thread: %s.",
+                    users_or_roles,
+                    thread,
+                )
+            return
+
+        if isinstance(users_or_roles, discord.Role):
+            member: discord.Member
+            for member in users_or_roles.members:
+                try:
+                    await thread.add_user(member)
+                except discord.NotFound:
+                    logger.debug(
+                        "User: %s has blocked the bot and "
+                        "therefore could not be added to thread: %s.",
+                        member,
+                        thread,
+                    )
+            return
+
+    @TeXBotBaseCog.listener()
+    @capture_guild_does_not_exist_error
+    async def on_thread_create(self, thread: discord.Thread) -> None:
+        """Add users to a thread when it is created."""
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        if not settings["ADD_COMMITTEE_TO_THREADS"]:
+            return
+
+        committee_role: discord.Role = await self.bot.committee_role
+        committee_elect_role: discord.Role = await self.bot.committee_elect_role
+
+        if thread.parent is None or thread.parent.category is None:
+            return
+
+        if "committee" in thread.parent.category.name.lower():
+            await self.add_users_or_roles_silently(
+                users_or_roles=[committee_role, committee_elect_role], thread=thread
+            )
```

Comment:
This event handler can be made less expensive by reordering the checks.
```suggestion
        committee_role: discord.Role = await self.bot.committee_role
        committee_elect_role: discord.Role = await self.bot.committee_elect_role

        if (
            thread.parent is None
            or thread.parent.category is None
            or "committee" not in thread.parent.category.name.lower()
            or not settings["ADD_COMMITTEE_TO_THREADS"]
        ):
            return

        await self.add_users_or_roles_silently(
            users_or_roles=[committee_role, committee_elect_role], thread=thread
        )
```